onnxruntime/core/framework/inline_containers.h

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cmath>
+#include <memory>
+#include <iostream>
+#include <memory_resource>
+#include <core/common/safeint.h>
+
+#include "core/framework/allocator.h"
+
+#pragma warning(push)
+#pragma warning(disable : 4127)
+#include <absl/container/inlined_vector.h>
+#include <absl/container/flat_hash_set.h>
+#include <absl/container/flat_hash_map.h>
+#pragma warning(pop)
+
+namespace onnxruntime {
+
+// Use InlinedVector for small arrays that can fit on a stack.
```

Comment:
It is a standard vector with a small buffer optimization. It would work in scenarios where small buffer is required most of the time, but occasionally we would need more memory and it would allocated it on the heap. std::array would not do it.